paymentsheet/src/main/java/com/stripe/android/paymentsheet/repositories/CustomerAdapter.kt

```diff
@@ -0,0 +1,91 @@
+package com.stripe.android.paymentsheet.repositories
+
+import androidx.annotation.RestrictTo
+import com.stripe.android.ExperimentalSavedPaymentMethodsApi
+import com.stripe.android.model.PaymentMethod
+
+/**
+ * [CustomerAdapter] A "bridge" from wallet mode to your backend to fetch Customer-related
+ * information. Typically, you will not need to implement this interface yourself. You should
+ * instead use [StripeCustomerAdapter], which implements [CustomerAdapter] and manages retrieving
+ * and updating a Stripe customer for you.
+ *
+ * Implement this interface if you would prefer retrieving and updating your Stripe customer object
+ * via your own backend instead of using [StripeCustomerAdapter].
+ */
+@ExperimentalSavedPaymentMethodsApi
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+interface CustomerAdapter {
+    /**
+     * Retrieves a list of payment methods attached to a customer
+     */
+    suspend fun fetchPaymentMethods(): Result<List<PaymentMethod>>
+
+    /**
+     * Adds a payment method to a customer
+     * @param paymentMethodId, the payment method to attach to the customer
+     */
+    suspend fun attachPaymentMethod(paymentMethodId: String)
+
+    /**
+     * Deletes the given payment method from the customer
+     * @param paymentMethodId, the payment method to detach from the customer
+     */
+    suspend fun detachPaymentMethod(paymentMethodId: String)
+
+    /**
+     * Set the selected payment method option.
+     * @param paymentOption, the [PersistablePaymentMethodOption] to save in the [DataStore]
+     */
+    suspend fun setSelectedPaymentMethodOption(paymentOption: PersistablePaymentMethodOption)
```

Comment:
I think this may need to be optional, because we can call `setSelectedPyamentMethodOption(paymentOption: nil)`, which should clear out the selectedPaymentMethodOption

Reply:
Makes sense to me. We should change the type, add docs (and eventually test)!